src/test_lib_json/main.cpp

```diff
@@ -307,7 +307,18 @@ JSONTEST_FIXTURE(ValueTest, arrayIssue252) {
   }
   // JSONTEST_ASSERT_EQUAL(5, root["array"].size());
 }
-
+JSONTEST_FIXTURE(ValueTest, arrayIssue691) {
+  Json::Value array2_;
+  array2_.append(1);
+  array2_.append(2);
+  array2_.append(3);
+  array2_.append(5);
+  ////use insert method
+  array2_.insert(3,4);
+  JSONTEST_ASSERT_EQUAL(Json::Value(3),array2_[2]);
```

Comment:
I would replace these checks with the following:

```cpp
  array2.append(10); // index 0
  array2.append(20); // index 1
  array2.append(30); // index 2
  array2.append(50); // index 3

  JSONTEST_ASSERT_EQUAL(Json::Value(50),array2_[3]);

  array2.insert(3, 40); // index 3

  // After updating, index 3 should be changed from 50 to 40, and 50 moved to index 4.
  JSONTEST_ASSERT_EQUAL(Json::Value(40),array2_[3]);
  JSONTEST_ASSERT_EQUAL(Json::Value(50),array2_[4]);

```

Reply:
Thanks a lot for your valuable suggestion.

Reply:
Hi @baylesj, the build still failed, but it's ok in my PC. I don't know why.

Reply:
Are you sure you just haven't updated the patch? The Mac, Windows, and Linux builds are all failing with the same error:

```
../src/lib_json/json_value.cpp:1183:20: error: too few arguments to function call, single argument 'index' was not specified
  if(!isValidIndex()){
      ~~~~~~~~~~~~ ^
../src/lib_json/json_value.cpp:1119:1: note: 'isValidIndex' declared here
bool Value::isValidIndex(ArrayIndex index) const { return index < size(); }
^
1 error generated.
```